enhancements/substitutes-for.md

```diff
@@ -0,0 +1,386 @@
+bundle which cannot be upgraded to the head of another channel).
+* Future bundles that reference the unpatched version of a bundle, when a patched version exists in the
+catalog, have their references updated to include the patched versions.
+* Clusters running a version of the bundle that predates the vulnerable bundle are not forced to
+upgrade into the vulnerable bundle before being able to upgrade to the patched bundle.
+* The code changes need to be contained within the catalog and opm binary so that it does not require
+OCP cluster changes.  It must work across existing 4.5+ versions of OCP clusters.
+
+### Non-Goals
+
+* Fully refactoring the upgrade graph management system out of bundle metadata so that upgrade graphs
+can be modified independently of bundle modification/publication.  (This is a future goal for OLM, however)
+
+
+## Proposal
+
+The proposal is to add a new field (tentatively named "substitutes-for") to CSVs, the value of which is one or
+more csv names that this bundle can be plugged in place of.  This will allow existing CSVs that indicate
+they "replace" or "skip" some other CSV, to also replace/skip the new CSV that substitutes for the one
+they skip/replace.  The net result of this is the ability to publish new CSVs that can still be upgraded
+to the latest CSV, without having to update the latest CSV to add new replaces/skips references.
+
+Because the set of substitutions may grow over time, and are not guaranteed to be added in any particular 
+order, the substitutions must be applied in an order independent way.  This means each time a new 
+substitution rule is added, additional substitution rules must be generated to account for transitive
+substitutions.
+
+For example, suppose we have the following set of relationships we need to add to the catalog
+(In this example, A has a CVE, but B has already shipped as a replacement for A.  We want to patch
+A as it is still the head of some channel.  We produce A' to patch A, and then find another issue in
+A' and produce A'' to patch A'):
+
+B replaces A (original CSV relationship)
+A' substitutes for A
+A'' substitutes for A'
+
+We would first add the rule:
+A' -> A
+
+Since this is the first entry, there are no additional rules to generate.
+
+Next we add:
+A'' -> A'
+
+Which generates the following transitive substitution(D->C->B):
+A'' -> A
+
+So the total set of substitution rules is:
+A' -> A
+A'' -> A'
+A'' -> A
+
+Which means that ultimately B will be able to replace A, A', or A'', as desired.
+
+
+
+If we work the example in a different order we would first add the rule:
+A'' -> A
+
+Since this is the first entry, there are no additional rules to generate.
+
+Next we add:
+A' -> A
+
+Which generates the following transitive substitution (D->C->B):
+A'' -> A
+
+Giving us the same total set of rules:
+A' -> A
+A'' -> A'
+A'' -> A
+
+Note that the generator for the new rules is the result of a simple inner join query
```

Comment:
It's not super clear to me that this needs to be a separate table. Can't this just use the existing skips mechanism?

(it might make sense to record the substites-for value per bundle, like we do for skips/replaces/skiprange)

Reply:
it doesn't necessarily have to be its own table, but i do think we need to store this info in its own column so we can apply the substitutions to future CSVs.